pkg/fetcher/fetcher.go

```diff
@@ -100,6 +100,21 @@ type URLFetcher struct {
 	options Options
 }
 
+// ErrorResponseBody struct
```

Comment:
Please expand upon this comment. Also, `PullErrorRespBody` would be a more suitable name for this struct.